### PR TITLE
Harvester: display more info on deleting namespace dialog

### PR DESCRIPTION
**Pull request by DaiYuzeng**
### Occurred changes and/or fixed issues

- https://github.com/harvester/harvester/issues/2591

### Areas or cases that should be tested

1. Go to Projects/Namespaces page
2. Remove one or more namespaces
3. The warning tips `Deleting the Namespace(s) will remove all the resources on this particular Namespace(s) and this is not revertable, are you sure you want to continue deleting XXX` should be visible on the dialog

image

### CODE REVIEW

File: shell/assets/translations/zh-hans.yaml [hunk 1]
```diff
@@ -3655,6 +3655,7 @@ promptRemove:
   attemptingToRemove: "你正在试图删除 {type}"
   protip: "提示：按住 {alternateLabel} 键的同时单击删除，以绕过此确认。"
   confirmName: "在下方输入 <b>{nameToMatch}</b> 以确认："
+  confirmRelatedResource: "删除 {type} 将会移除 {type} 有关的所有资源，并且该操作不可逆，请确认是否继续删除"
```
Comment (WuJun2016) [inline shell/assets/translations/zh-hans.yaml:3658]: Inconsistency between Chinese translation and English

Comment (WuJun2016) [reply shell/assets/translations/zh-hans.yaml:3658]: 删除命名空间将删除这个特定命名空间上的所有资源，这是不可恢复的，您确定要继续删除吗 ？

File: shell/components/dialog/ConfirmRelatedToRemoveDialog.vue [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+      <AsyncButton mode="delete" class="btn bg-error ml-10" :disabled="deleteDisabled" @click="remove" />
+    </template>
+  </Card>
+</template>
+
+<style lang='scss' scoped>
+  .actions {
+    text-align: right;
+  }
+</style>
```
Comment (WuJun2016) [inline shell/components/dialog/ConfirmRelatedToRemoveDialog.vue:157]: Components are not universal, later on other list pages may also need similar tips.

**Review by WuJun2016 — CHANGES_REQUESTED** [review]

File: shell/components/dialog/ConfirmRelatedToRemoveDialog.vue [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+      required: true
+    }
+  },
+
+  data() {
+    return { errors: [], confirmName: '' };
+  },
+
+  computed: {
+    ...mapState('action-menu', ['modalData']),
```
Comment (n313893254) [inline shell/components/dialog/ConfirmRelatedToRemoveDialog.vue:41]: Seems `action-menu` unused?

**Review by n313893254 — APPROVED** [review]
LGTM, I noticed `ConfirmRelatedToRemoveDialog` is similar to `ConfirmNameToRemoveDialog`, I think it would be better to move to `dialog/harvester` if the component is only designed for Harvester. Otherwise, better extend or add property based on `ConfirmNameToRemoveDialog`

File: shell/components/dialog/ConfirmRelatedToRemoveDialog.vue (hunk 3 above)
Comment (DaiYuzeng) [reply shell/components/dialog/ConfirmRelatedToRemoveDialog.vue]: I think `action-menu` like the module/namespace of `modalData`, you cannot get modalData state if you do not declare the module/namespace

**Review by n313893254 — APPROVED** [review]
LGTM

**Review by WuJun2016 — APPROVED** [review]
LGTM

**Pull request merged**